core/filters/content_safety.py

```
@@ -0,0 +1,218 @@
+
+def openai_chat_url(base_url: str, allow_private_endpoint: bool = False) -> str:
+    url = str(base_url or "").strip().rstrip("/")
+    parsed = urlparse(url)
+    if parsed.scheme not in {"http", "https"} or not parsed.hostname:
+        raise ValueError("AI Base URL 必须是完整的 HTTP(S) 地址")
+    try:
+        address = ipaddress.ip_address(parsed.hostname)
+    except ValueError:
+        address = None
+    if address is not None and not allow_private_endpoint:
+        raise ValueError("AI Base URL 不允许使用 IP 地址，请显式开启私网端点")
```

Ruff (0.15.21)

[warning] String contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF001)